### ISSUE TITLE
Wysihat fails to load in IE7

### ISSUE THREAD

**Issue by keeran** [80beans/wysihat-engine#3]
Have tried to get the demo app running in IE7. It fails with a Javascript 'object expected' error.

Have run it through the Visual Web Developer debugger (only way I could get something useful!) and it breaks at the point where 'addButtons' is called:

   <label for="page_content">Content</label><br />
    <script type="text/javascript">addButtons('page_content', ['bold', 'italic', 'underline', 'strikethrough', 'justify_left', 'justify_center', 'justify_right', 'insert_ordered_list', 'insert_unordered_list', 'undo', 'redo', 'link', 'html', 'image']);

Still digging around to work out why - initial guess is that this has something to do with the latest change to make the JS unobtrusive.

**Comment by jeffkreeftmeijer** [comment]
Hi Kee, 

Those nasty IE javascript errors usually have something to do with stray commas and forgotten semicolons. I've updated the wysihat_engine.js file and if you ask me the whole thing is nice and tidy now.

I've pushed a fix and updated the gem on gemcutter. Also, I've updated the demo, so it's a good idea to just clone that and see if it works :)

I haven't checked this in IE yet (and I'm not even entirely sure that this was the problem), but I think this will solve your issue. I will check it out as soon as possible.

Update:

I found myself a beige box and checked it out. The problem was that the addButton calls got called before the toolbar was created. That was odd, because it only happened in IE.

The creating of the editor and adding of the buttons were divided into two functions.

Since they were called at the same time I decided to put them both into one function and call that one instead of the addButtons one in previous versions. This function is called wysiHatify().

Now IE doesn't have a problem trying to figure out which one to call first anymore, since there's only one.

This should be fixed now. Thanks for reporting! :)
